cffi/cdefs.h

```diff
@@ -613,6 +613,25 @@ struct lysp_node_list {
     ...;
 };
 
+struct lysp_node_grp {
```

Comment:
Could you add this in the commit where it is used?

Reply:
This is not used directly in the code of libyang-python. I am traversing "parsed" tree in my code directly by accessing `Module.cdata.parsed` and looking specifically for groupings within the tree. I can read many parsed structures already, but not groupings.

I thought about creating similar structure for "parsed" tree as SNode is for "compiled", but that is much more complex job, which I am not sure is needed.

Reply:
I don't think we should add cdefs that are not used in the library itself. @rjarry, thoughts?

Reply:
I agree with @samuel-gauthier, please don't add cdefs if they are not used by the python library. We don't want to expose the cffi api to external users.

However, if you want to expose the groupings to the API, we need at least a basic wrapper in python. Would that be something you could do?

Reply:
I have updated the code quite massively to support traversing over parsed schema tree. Hopefully this would be ok... suggestion for future would be to replace to separate `SNode` based code from `PNode` code to avoid and if `SNode` need to use `PNode`, create `PNode` and forward the request. This would avoid having duplicate code and having `SNode` working with just specific compile data as it should be. Currently it is kind of mixture. The same goes for `Module` and `PModule` (which I didn't create)